### PR TITLE
Remove form config backend - #447

### PR DESCRIPTION
**Pull request by jozefbakus**
Implements: 
- #449 
- #447 
- #448

Changes: 
- Removed generic forms
- Created components for each job/sensor/template type
- Removed dynamic form configuration
- Refactored form parts (removed part validation, propagating changes using @output)
 - Removed form config.
 - Only 2 job/template types in Backend: Spark and Shell
 - Implemented validation on mandatory fields

### CODE REVIEW

File: src/main/resources/db_scripts/liquibase/v0.5.2.remove-form-config.sql [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            else updateJobType.jobDefinition::jsonb
+            end
+        ) as jobDefiniton
+    from updateJobType
+    group by id
+)
+update workflow_history
+set workflow = jsonb_set(workflow::jsonb, array['dagDefinitionJoined', 'jobDefinitions'], updatedJobTemplateId.jobDefiniton)
+FROM updatedJobTemplateId
+where updatedJobTemplateId.id=workflow_history.id;
```
Comment (kevinwallimann) [inline src/main/resources/db_scripts/liquibase/v0.5.2.remove-form-config.sql]: Please add a newline at the end of the file

File: src/main/resources/db_scripts/liquibase/v0.5.2.remove-form-config.sql [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+alter table job_template rename column form_config to form_config_old;
+
+update job_definition
+set job_parameters = job_parameters - 'formConfig' || jsonb_build_object('jobType', job_parameters->'formConfig');
+
+alter table job_definition alter column job_template_id drop not null;
+
+update job_definition set job_template_id = NULL
+where
+    ((job_parameters::jsonb ->> 'jobType' = 'Shell') AND (job_parameters::jsonb ->> 'scriptLocation' IS NOT NULL)) OR
```
Comment (kevinwallimann) [inline src/main/resources/db_scripts/liquibase/v0.5.2.remove-form-config.sql]: Why do you need this condition? `AND (job_parameters::jsonb ->> 'scriptLocation' IS NOT NULL)`

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationService.scala [hunk 3]
```diff
@@ -167,4 +168,29 @@ class WorkflowValidationServiceImpl @Inject()(override val workflowRepository: W
...
+            Future.successful(Seq.empty[ApiError])
+          case ShellDefinitionParameters(jobType, scriptLocation) if scriptLocation.isEmpty =>
+            Future.successful(Seq(ValidationError(s"Script location cannot be empty in case of $jobType job type when template is empty")))
+          case ShellDefinitionParameters(_, _) =>
+            Future.successful(Seq.empty[ApiError])
+        }
+        case Some(jobTemplateId) => jobTemplateService.getJobTemplate(jobTemplateId).map { jobTemplate =>
+          if (jobTemplate.jobParameters.jobType != jobDefinition.jobParameters.jobType)
+            Seq(ValidationError(
+              s"Template's job type has to be the same as job's job type. Template - ${jobTemplate.jobParameters.jobType} is not equlat to Job - ${jobDefinition.jobParameters.jobType}"
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationService.scala]: Typo `equlat`

File: src/test/scala/za/co/absa/hyperdrive/trigger/ApplicationStartPostgresTest.scala [hunk 4]
```diff
@@ -88,11 +88,24 @@ class ApplicationStartPostgresTest extends FlatSpec with Matchers with SpringInt
     notificationConfig.enabled shouldBe true
 
     hyperDriverManager.isManagerRunning shouldBe true
+    val jobTemplateSpark = JobTemplateFixture.GenericSparkJobTemplate
+    val jobTemplateShell = JobTemplateFixture.GenericShellJobTemplate
+    await(jobTemplateRepository.insertJobTemplate(jobTemplateSpark))
+    await(jobTemplateRepository.insertJobTemplate(jobTemplateShell))
+    val jobTemplates = await(jobTemplateRepository.getJobTemplates())
+
     val workflowJoined = WorkflowFixture.createWorkflowJoined()
```
Comment (kevinwallimann) [inline src/test/scala/za/co/absa/hyperdrive/trigger/ApplicationStartPostgresTest.scala]: If you want, you can also create two parameters for the template ids in `createWorkflowJoined`, instead of doing this copy-gymnastics

File: src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationServiceTest.scala [hunk 5]
```diff
@@ -117,13 +139,60 @@ class WorkflowValidationServiceTest extends AsyncFlatSpec with Matchers with Moc
     )
   }
 
+  it should "fail if job template and job definition types are not equal" in {
```
Comment (kevinwallimann) [inline src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationServiceTest.scala:142]: There should also be a test for `Job jar and main class cannot be empty in case of $jobType job type when template is empty`

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateResolutionService.scala [hunk 6]
```diff
@@ -86,7 +87,7 @@ class JobTemplateResolutionServiceImpl extends JobTemplateResolutionService {
 
   private def mergeSparkParameters(definitionParams: SparkDefinitionParameters, templateParams: SparkTemplateParameters): SparkInstanceParameters = {
     SparkInstanceParameters(
-      jobType = definitionParams.jobType,
+      jobType = JobTypes.Spark,
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateResolutionService.scala:90]: Can't you put `jobType = JobTypes.Spark` into the `SparkInstanceParameters` case class as a default parameter, like for `ShellInstanceParameters`?

Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateResolutionService.scala]: Can't you put `jobType = JobTypes.Spark` into the `SparkInstanceParameters` case class as a default parameter, like for `ShellInstanceParameters`?

File: src/main/resources/db_scripts/liquibase/v0.5.2.remove-form-config.sql (hunk 2 above)
Comment (jozefbakus) [reply src/main/resources/db_scripts/liquibase/v0.5.2.remove-form-config.sql]: Because if it is null it should leave template id. 
We don't have such templates, but it is possible.
It is possible to have spark or shell template, which theoretically can provide scriptLocation.

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateResolutionService.scala (hunk 6 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateResolutionService.scala]: Done

File: src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationServiceTest.scala (hunk 5 above)
Comment (jozefbakus) [reply src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationServiceTest.scala:142]: Done

File: src/test/scala/za/co/absa/hyperdrive/trigger/ApplicationStartPostgresTest.scala (hunk 4 above)
Comment (jozefbakus) [reply src/test/scala/za/co/absa/hyperdrive/trigger/ApplicationStartPostgresTest.scala]: Done

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationService.scala (hunk 3 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationService.scala]: Fixed

File: src/main/resources/db_scripts/liquibase/v0.5.2.remove-form-config.sql (hunk 1 above)
Comment (jozefbakus) [reply src/main/resources/db_scripts/liquibase/v0.5.2.remove-form-config.sql]: OK

**Review by kevinwallimann — DISMISSED** [review]

**Review by kevinwallimann — APPROVED** [review]
